src/burnchains/burnchain.rs

```diff
@@ -1213,12 +1213,21 @@ impl Burnchain {
         );
 
         if let Some(target_block_height) = target_block_height_opt {
```

Comment:
So, there's no test? How was this change tested and why do we know it is an improvement?

Reply:
+1

Reply:
So, no, unfortunately there's no test. 
This change was tested locally with a copy the chainstate of a miner that was stuck because of this bug, and this branch was then directly deployed by @CharlieC3 to the miner on the same chainstate, and that helped us unblocking the miner - looks like an improvement. 

The code coverage for this routine is pretty bad because this function is doing way too many things. 
I think we could rewrite this routine, extract the logic dealing with the boundaries and target block (there are a lot of `if` in there that would deserve some testing), and unit test that logic.

Reply:
OK. Can you just summarize this in the PR description?